### PR TITLE
icingacli x509 check: take the whole chain into account

### LINKED ISSUE #64: Validation does not evaluate the entire chain
## Describe the bug
The check will not show WARNING or CRITICAL if the server is presenting a full chain with
'remaining time of ca/intermediate less than leaf certificate'.

the following example is showing an example of the certificate usage in that case and reports 'OK - .... expires in 524 days'
image

## To Reproduce

import some certificate chain with specific data. essential example data is given below in the table in section 'Additional context'.

background: cross signing certificates often used to transform CAs. In such situations it might happen that more than one certificate path is possible.
For details see the current case with sectigo certificates: https://support.sectigo.com/Com_KnowledgeDetailPageFaq?Id=kA01N000000rgSZ

## Expected behavior
CheckCommand should evaluate the whole chain.

## Screenshots
this is shown in the description above:
image

## Your Environment
This is an implementation bug it will happen in any version i checked.

* Icinga Web 2 version and modules (System - About):
* Web browser used:
* Icinga 2 version used (`icinga2 --version`):
* PHP version used (`php --version`):
* Server operating system and version:

## Additional context

I modified the query from the check slightly to show case more insights of this problem.
```
SELECT c.id,
       t.port,
       cc.valid,
       cc.invalid_reason,
       c.subject,
       ci.self_signed as ci_ss,
       c.self_signed as c_ss,
       COALESCE(ci.self_signed, c.self_signed) as self_signed,
       c.valid_from,
       c.valid_to,
       ccl.`order`
FROM x509_target t
         JOIN x509_certificate_chain cc ON cc.id = t.latest_certificate_chain_id
         JOIN x509_certificate_chain_link ccl ON ccl.certificate_chain_id = cc.id
         JOIN x509_certificate c ON c.id = ccl.certificate_id
         LEFT JOIN x509_certificate ci ON ci.subject_hash = c.issuer_hash
WHERE t.hostname = 'hallo.mywildcard.com' and
      t.port = 443
ORDER BY ccl.`order` ASC;
```

| id | port | valid | invalid\_reason | subject | ci\_ss | c\_ss | self\_signed | valid\_from | valid\_to | order |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| 40 | 443 | yes | NULL | \*.mywildcard.com | no | no | no | 1558137600 | 1621295999 | 0 |
| 10 | 443 | yes | NULL | Sectigo RSA Domain Validation Secure Server CA | no | no | no | 1541116800 | 1924991999 | 1 |
| 10 | 443 | yes | NULL | Sectigo RSA Domain Validation Secure Server CA | yes | no | yes | 1541116800 | 1924991999 | 1 |
| 41 | 443 | yes | NULL | USERTrust RSA Certification Authority | yes | no | yes | 959683718 | 1590835718 | 2 |
| 36 | 443 | yes | NULL | AddTrust External CA Root | yes | yes | yes | 959683718 | 1590835718 | 3 |

that shows some more droubles here with the `join` and maybe the 2 cert chain pathes 
see https://support.sectigo.com/Com_KnowledgeDetailPageFaq?Id=kA01N000000rgSZ

### PR DESCRIPTION
**Pull request by Al2Klimov**
not just the leaf certificate while calculating the expire time for the case an intermediate CA expires sooner than the leaf. In the latter case the check command warns earlier now.

fixes #64

### CODE REVIEW

**Comment by Al2Klimov** [comment]
## Before

```
$ icingacli x509 check --ip 127.0.0.1 --port 4433
OK - 127.0.0.1 expires in 399 days|'127.0.0.1'=34492225s;8640000;3456000;0;34560000
```

## After

```
$ icingacli x509 check --ip 127.0.0.1 --port 4433
OK - 127.0.0.1 expires in 199 days|'127.0.0.1'=17210893s;4320000;1728000;0;17279999
```

File: application/clicommands/CheckCommand.php [hunk 1]
```diff
@@ -72,8 +72,16 @@ public function hostAction()
                 'cc.invalid_reason',
                 'c.subject',
                 'self_signed'   => 'COALESCE(ci.self_signed, c.self_signed)',
-                'c.valid_from',
-                'c.valid_to'
+                'valid_from'    => (new Select())
+                    ->from('x509_certificate_chain_link xccl')
+                    ->columns('MAX(xc.valid_from)')
+                    ->join('x509_certificate xc', 'xc.id = xccl.certificate_id')
+                    ->where('xccl.certificate_chain_id = cc.id'),
```
Comment (yhabteab) [inline application/clicommands/CheckCommand.php:79]: Well, it didn't work for me. However, I modified the query a bit and got it.

### Before

```bash
root@a2f6d2aaab32:/# icingacli x509 check --ip 10.211.55.31 --port 443
OK - Icinga Web 2 expires in 14 days|'Icinga Web 2'=1241677s;345600;172800;0;1296000
```

### After

```bash
root@a2f6d2aaab32:/# icingacli x509 check --ip 10.211.55.31 --port 443
OK - Icinga Web 2 expires in 4 days|'Icinga Web 2'=377696s;86400;0;0;424139
```

```suggestion
                    ->from('x509_certificate xc')
                    ->columns('MAX(xci.valid_from)')
                    ->join('x509_certificate xci', 'xci.issuer_hash = xc.issuer_hash')
                    ->where('xc.id = c.id'),
```

**Comment by yhabteab** [comment]
Bildschirmfoto 2022-09-15 um 09 09 03

Bildschirmfoto 2022-09-15 um 09 10 12

**Review by yhabteab — APPROVED** [review]
Tested also with PGSQL.

```bash
root@fcda98f615fc:/# icingacli x509 check --ip 10.211.55.31 --port 443
OK - Icinga Web 2 expires in 4 days|'Icinga Web 2'=369996s;86400;0;0;431902
```

**Comment by Al2Klimov** [comment]
And I with intermediate certs. Got 97 days validity now. ✅

**Review by lippserd — APPROVED** [review]

**Pull request merged**